packages/clerk-js/src/ui/hooks/usePopover.ts

```diff
@@ -22,14 +23,36 @@ export type UsePopoverReturn = ReturnType<typeof usePopover>;
 export const usePopover = (props: UsePopoverProps = {}) => {
   const { bubbles = true } = props;
   const [isOpen, setIsOpen] = React.useState(props.defaultOpen || false);
+  const nodeId = useFloatingNodeId();
   const { update, reference, floating, strategy, x, y, context, refs } = useFloating({
     open: isOpen,
     onOpenChange: setIsOpen,
+    nodeId,
     whileElementsMounted: props.autoUpdate === false ? undefined : autoUpdate,
     placement: props.placement || 'bottom-start',
     middleware: [offset(props.offset || 6), flip(), shift()],
   });
 
+  useDismiss(context, { bubbles });
+
+  useEffect(() => {
+    const handleFocus = (e: FocusEvent) => {
+      if (e.relatedTarget === null) {
```

Comment:
❓ Is there a reason we're explicitly checking for === null instead of !e.relatedTarget?

Reply:
Don't think so. Will change it.